bundles/org.openhab.binding.pushover/src/main/java/org/openhab/binding/pushover/internal/handler/PushoverAccountHandler.java

```diff
@@ -0,0 +1,163 @@
+import org.openhab.binding.pushover.internal.config.PushoverAccountConfiguration;
+import org.openhab.binding.pushover.internal.config.PushoverConfigOptionProvider;
+import org.openhab.binding.pushover.internal.connection.PushoverAPIConnection;
+import org.openhab.binding.pushover.internal.connection.PushoverCommunicationException;
+import org.openhab.binding.pushover.internal.connection.PushoverConfigurationException;
+import org.openhab.binding.pushover.internal.connection.PushoverMessageBuilder;
+import org.openhab.binding.pushover.internal.dto.Sound;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link PushoverAccountHandler} is responsible for handling commands, which are sent to one of the channels.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class PushoverAccountHandler extends BaseThingHandler {
+
+    private static final Collection<Class<? extends ThingHandlerService>> SUPPORTED_THING_ACTIONS = Set
+            .of(PushoverActions.class, PushoverConfigOptionProvider.class);
+
+    private final HttpClient httpClient;
+
+    private @NonNullByDefault({}) PushoverAccountConfiguration config;
+    private @Nullable PushoverAPIConnection connection;
+
+    public PushoverAccountHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // nothing
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(PushoverAccountConfiguration.class);
+
+        boolean configValid = true;
+        final String apikey = config.apikey;
+        if (apikey == null || apikey.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error-missing-apikey");
+            configValid = false;
+        }
+        final String user = config.user;
+        if (user == null || user.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error-missing-user");
+            configValid = false;
+        }
+
+        if (configValid) {
+            updateStatus(ThingStatus.UNKNOWN);
+
+            connection = new PushoverAPIConnection(httpClient, config);
+            try {
+                connection.validateUser();
+                updateStatus(ThingStatus.ONLINE);
+            } catch (PushoverCommunicationException | PushoverConfigurationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
```

Comment:
`connection.validateUser` might be slow, so please make this code asynchronous so the `initialize()` method remains speedy.

Reply:
Good catch. I somehow had in my mind I planned to do that.